src/proxy/processors/push-action/checkSensitiveData.js

```diff
@@ -0,0 +1,171 @@
+            if (err) {
+                console.error(`Error reading file ${filePath}: ${err.message}`);
+                return reject(err);
+            }
+            if (checkForSensitiveData(data)) {
+                console.log(`\x1b[Sensitive data found in ${filePath}\x1b[0m`);
+                sensitiveDataFound = true;
+            }
+            resolve(sensitiveDataFound);
+        });
+    });
+};
+// Function to parse the file based on its extension
+const parseFile = async (filePath) => {
+  
+    const ext = path.extname(filePath).toLowerCase();
+    const FilestoCheck = commitConfig.diff.block.proxyFileTypes;
+    if(!FilestoCheck.includes(ext)){
+      
+        console.log(`${ext} should be included in CommitConfig for proxy Check!`);
+        return false;
+    }
+    
+    switch (ext) {
+        case '.csv':
+            return await processCSV(filePath);
+        case '.xlsx':
+            return await processXLSX(filePath);
+        case '.log':
+            return await checkLogJsonFiles(filePath);
+        case '.json':
+            return await checkLogJsonFiles(filePath);
+        default:
+            // Skip unsupported file types without logging
+            return false; // Indicate that no sensitive data was found for unsupported types
+    }
+};
+// Async exec function to handle actions
+// Function to parse file paths from git diff content
+const extractFilePathsFromDiff = (diffContent) => {
+    const filePaths = [];
+    const lines = diffContent.split('\n');
+    
+    lines.forEach(line => {
+        const match = line.match(/^diff --git a\/(.+?) b\/(.+?)$/);
+        if (match) {
+            filePaths.push(match[1]); // Extract the file path from "a/" in the diff line
+        }
+    });
+    
+    return filePaths;
+};
+
+const exec = async (req, action) => {
+    const diffStep = action.steps.find((s) => s.stepName === 'diff');
+    const step = new Step('checksensitiveData');
+
+    if (diffStep && diffStep.content) {
+        console.log('Diff content:', diffStep.content);
+
+        // Use the parsing function to get file paths
+        const filePaths = extractFilePathsFromDiff(diffStep.content);
+
+        if (filePaths.length > 0) {
+            // Check for sensitive data in all files
+            const sensitiveDataFound = await Promise.all(filePaths.map(parseFile));
+            const anySensitiveDataDetected = sensitiveDataFound.some(found => found);
+
+            if (anySensitiveDataDetected) {
+                step.blocked= true;
+                step.error = true;
+                step.errorMessage = 'Your push has been blocked due to sensitive data detection.';
+                console.log(step.errorMessage);
+            }
+        } else {
+            console.log('No file paths provided in the diff step.');
+        }
+    } else {
+        console.log('No diff content available.');
+    }
+    action.addStep(step);
+    return action; // Returning action for testing purposes
+};
```

Comment:
@Psingle20 since #793 has been released in 1.6.0, can we restructure this functionality into its own plugin? It'll require moving some files around and creating an npm package using `npm init`.

The other change will be this:
```
const Step = require('@finos/git-proxy/src/proxy/actions').Step;
const config = require('@finos/git-proxy/src/config');
```

Use [plugins/git-proxy-sample-plugins](https://git-proxy.finos.org/docs/development/plugins) and refer to the docs (to be improved via #811) for details.

Reply:
I will do that. Btw we have added some refactor in #810  which also contains some more feature should I do a PR individually for all the plugins ?
ALong with this we have added gitleaks support , EXIF metadata check and AIML usage check.